### PR TITLE
Sparsity-Promoting DMD

### LINKED ISSUE #105: Sparsity Promoting DMD
Hi,

Are there plans to incorporate sparsity promoting DMD? 

Adding an option to post-process the modes using a convexopt procedure to identify modes? If truncation is sufficient, is there a good discussion for why?

If there is a need, and help is needed, happy to consider helping over the next few months when I have time.

Reference: http://people.ece.umn.edu/users/mihailo/papers/jovschnicPOF14.pdf

### PR DESCRIPTION
**Pull request by fAndreuzzi**
In this commit I added an implementation of SPDMD (https://aip.scitation.org/doi/10.1063/1.4863670), along with tests, documentation of functions and a small tutorial using an heat transfer dataset.

This PR fixes #105.

I also used NumPy's [function](https://numpy.org/doc/stable/reference/generated/numpy.vander.html) to compute Vandermonde's matrix (#95b59a7).

The results of this implementation have been tested against does provided by the implementation which I found [here](http://www.pyrunner.com/weblog/2016/08/03/spdmd-python/).

### CODE REVIEW

**Comment by mtezzele** [comment]
I think you should remove this file: tutorials/tutorial-10-spdmd.ipynb
Moreover please squash some commits.

**Comment by fAndreuzzi** [comment]
> I think you should remove this file: tutorials/tutorial-10-spdmd.ipynb
This is the tutorial I wrote on using SPDMD.

> Moreover please squash some commits.
Sure

File: pydmd/spdmd.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+            old_beta = beta
+            alpha, beta, lmbd = self._update(beta, lmbd)
+
+        if self._verbose:
+            print("ADMM: {} iterations".format(i))
+
+        # zero values in beta are associated with DMD amplitudes which can be
+        # set to 0
+        return np.abs(old_beta) < 1.0e-12
```
Comment (ndem0) [inline pydmd/spdmd.py:283]: Can we have a settable tolerance? For some problems, I suppose it makes slight difference

**Review by ndem0 — COMMENTED** [review]
- There are two different files `ipynb` for the tutorial, please remove one of them (tutorial-10-spdmd.ipynb I would say)
- Please adjust the README in tutorials folder with the new tutorial
- Please add the html version to the `docs/source/_tutorials`
- Please adjust the `docs/source/index.rst` with the new tutorial

**Comment by fAndreuzzi** [comment]
Thank you @ndem0 and @mtezzele, I'm going to work on this PR to address the problems you raised.

**Pull request closed without merging**